### PR TITLE
[ROAD-709] Fix: snyk code cache issue

### PR DESCRIPTION
**Pull request by aldanchenko**
In this PR:

- fix SnykCode cache issue.
- fix NullReferenceException issue.

### CODE REVIEW

File: Snyk.Common.Tests/Snyk.Common.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
+    <PackageReference Include="coverlet.collector" Version="3.0.2">
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\Snyk.Code.Library\Snyk.Code.Library.csproj" />
```
Comment (michelkaporin) [inline Snyk.Common.Tests/Snyk.Common.Tests.csproj:24]: Common shouldn't have any reference to other projects.

File: Snyk.Common.Tests/FileProviderTest.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+namespace Snyk.Common.Tests
```
[the remark this answers is not in the document]
Comment (aldanchenko) [reply Snyk.Common.Tests/FileProviderTest.cs:1]: @michelkaporin I also think about this. But IFileProvider belongs to Snyk.Common project. Or?

File: Snyk.Code.Library/Service/SnykCodeFileProvider.cs [hunk 3]
```diff
@@ -62,6 +62,8 @@ public void ClearHistory()
         {
             this.changedFiles.Clear();
             this.removedFiles.Clear();
+
+            this.solutionPath = null;
```
[the remark this answers is not in the document]
Comment (aldanchenko) [reply Snyk.Code.Library/Service/SnykCodeFileProvider.cs:66]: In this method we initialise this - https://github.com/snyk/snyk-visual-studio-plugin/blob/fb3bbebcd446a00391ed0a60c813ea36379f07dc/Snyk.Code.Library/Service/SnykCodeFileProvider.cs#L36

If user open one project and after this open another we need to reset this value.

File: Snyk.Common.Tests/FileProviderTest.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (michelkaporin) [reply Snyk.Common.Tests/FileProviderTest.cs:1]: Correct, but you're testing implementation, not the interface, right?

**Review by michelkaporin — APPROVED** [review]
👍

**Pull request merged**